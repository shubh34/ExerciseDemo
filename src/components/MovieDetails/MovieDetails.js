import React from 'react';

const MovieDetails = () => <div>this is MovieDetails page </div>;

export default MovieDetails;
