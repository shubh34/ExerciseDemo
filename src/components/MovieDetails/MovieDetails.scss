.movie-details {
    @media screen and (min-width: 1140px) {
        padding-top: 60px;
    }
}

.movie-info {
    position: relative;
    height: 100%;
}
.details_actions {
    position: relative;
    top: 30%;
   
}
.details_meta {
    position: absolute;
    bottom: -16px;
    display: block;
    width: 100%
}
.details_meta_title {
    text-align: center;
}
.movie-details-description {
    padding: 16px;
    @media screen and (min-width: 1140px) {
        padding-top: 24px;
    }
    letter-spacing: 0.5px;
    line-height: 32px;
}
