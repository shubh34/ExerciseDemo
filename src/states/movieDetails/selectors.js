import get from 'lodash/get';
import { createSelector } from 'reselect';

const getMovieDetails = state => get(state, 'movieDetails', '');

export const getHighLightedScore = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.scores', []).find(score => score.highlighted));

export const getMovieTitle = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.title', ''));

export const getSnapshotImageUrl = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.images.snapshot', '').replace('.jpeg', '-width1920-quality80.jpeg'));

export const getMovieDescription = createSelector(getMovieDetails, movieDetails => get(movieDetails, 'data.plot', ''));

export const getScore = createSelector(getHighLightedScore, scores => get(scores, 'score', ''));

export const getVotes = createSelector(getHighLightedScore, scores => get(scores, 'formatted_amount_of_votes', ''));
