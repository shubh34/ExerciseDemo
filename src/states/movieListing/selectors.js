import get from 'lodash/get';

export const getMovieListHeader = (state, id) => get(state, `movieListing[${id}].data.name`);

export const getMovieListByCategory = (state, id) => get(state, `movieListing[${id}].data.contents.data`, []).map(movie => ({
	artwork: movie.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
	id: movie.id,
	score: movie.highlighted_score.score,
	votes: movie.highlighted_score.formatted_amount_of_votes,
}));
