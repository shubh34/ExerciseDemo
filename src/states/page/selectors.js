export const getCurrentPage = state => state.page.name;
export const getBackPage = state => state.page.backPage;
