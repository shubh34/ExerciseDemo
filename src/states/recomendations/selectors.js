import get from 'lodash/get';
import { createSelector } from 'reselect';

export const getRecommendedMovies = state => get(state, 'recomendations.data', []);

export const getFormattedRecomendedMovieDetails = createSelector(
	getRecommendedMovies,
	movies => movies.map(movieDetails => ({
		id: movieDetails.id,
		artwork: movieDetails.images.artwork.replace('.jpeg', '-width217-quality80.jpeg'),
		score: movieDetails.highlighted_score.score,
		votes: movieDetails.highlighted_score.formatted_amount_of_votes
	}))
);
