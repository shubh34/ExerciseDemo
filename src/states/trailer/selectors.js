import get from 'lodash/get';

export const getMovieTrailerUrl = state => get(state, 'trailer.trailer.data.stream_infos[0].url');
