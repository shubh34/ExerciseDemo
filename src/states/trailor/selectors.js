import get from 'lodash/get';

export const getMovieTrailorUrl = state => get(state, 'trailor.trailor.data.stream_infos[0].url');
