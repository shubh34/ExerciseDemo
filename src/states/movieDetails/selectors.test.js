import { getMovieTitle, getSnapshotImageUrl, getMovieDescription, getVotes, getScore } from './selectors';
import { movieDetailsStub } from '../../../tests-stub/MovieDetails';

const state = { movieDetails: movieDetailsStub };

describe('MovieListing Selector', () => {
	it('should return getMovieTitle', () => {
		expect(getMovieTitle(state)).toMatchSnapshot();
	});
	it('should return snapshot url ', () => {
		expect(getSnapshotImageUrl(state)).toMatchSnapshot();
	});

	it('should return details', () => {
		expect(getMovieDescription(state)).toMatchSnapshot();
	});

	it('should return votes', () => {
		expect(getVotes(state)).toMatchSnapshot();
	});
	it('should return score', () => {
		expect(getScore(state)).toMatchSnapshot();
	});
});
