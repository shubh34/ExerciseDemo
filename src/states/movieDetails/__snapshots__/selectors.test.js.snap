// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListing Selector should return details 1`] = `"Capitana Marvel te lleva a una espectacular aventura en los años 90, siguiendo el camino de Carol Danvers (Brie Larson) mientras se convierte en una de las heroínas más poderosas del universo. Cuando la guerra galáctica llega a la Tierra, se encuentra con el agente Nick Furia (Samuel L. Jackson) en el centro del caos, que le lleva a su destino final como Vengador."`;

exports[`MovieListing Selector should return getMovieTitle 1`] = `"Capitana Marvel"`;

exports[`MovieListing Selector should return score 1`] = `7`;

exports[`MovieListing Selector should return snapshot url  1`] = `"https://images-3.wuaki.tv/system/shots/144189/original/capitana-marvel-1560352960-width1920-quality80.jpeg"`;

exports[`MovieListing Selector should return votes 1`] = `"6,5K"`;
